docs/en/03_configuration/31_adding_data_to_an_index.md

```diff
@@ -0,0 +1,30 @@
+# Adding data to an index
+
+Once you have [created your index](./30_creating_an_index.md), you can add data to it in a number of ways.
+
+## Reindex the site
+
+Running the [Solr reindex task](./33_dev_tasks.md) will crawl your site for classes that match those defined on your index, and add the defined fields to the index for searching. This is the most common method used to build the index the first time, or to perform a full rebuild of the index.
+
+## Publish a page in the CMS
+
+Every change, addition or removal of an indexed class instance triggers an index update through a "processor" object. The update is transparently handled through inspecting every executed database query and checking which database tables are involved in it.
+
+A reindex event will trigger when you make a change in the CMS, via `SearchUpdater::handle_manipulation()`, or `ProxyDBExtension::updateProxy()`. This tracks changes to the database, so any alterations will trigger a reindex. In order to minimise delays to those users, the index update is deferred until after the actual request returns to the user, through PHP's `register_shutdown_function()` functionality.
+
+## Manually
+
+If you get desperate, you can create a new page in a build task or something like that:
```

Comment:
Hmm, maybe a bit more professional/confident wording? "desparate" and "something like that" are a bit weak.